ls1intum/Artemis
# `Quiz exercises`: Fix synchronized mode due date validation

### Checklist
#### General


#### Client

### Motivation and Context



Closes #5772

### Description

For a quiz whose mode is synchronized, the due date was set to start date + duration + grace period (5 seconds) on the server side as implemented [here](https://github.com/ls1intum/Artemis/blob/b27d53d4d7a543f4849d1e2b628d22430a53c690/src/main/java/de/tum/in/www1/artemis/service/QuizExerciseService.java#L168). When editing the quiz, the due date is never modified even though the start date is changed. Therefore, it is possible that the release date and start date are modified in such a way that: due date < release date < start date. This caused the quiz date to be invalid. To prevent the situation, the due date is set to be undefined (for synchronized mode), so that the quiz date is considered valid.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course
- 1 Quiz Exercise whose mode is synchronized

1. Log in to Artemis
2. Navigate to Course Administration
3. Select the Course
4. Click Exercises
5. Edit the Quiz Exercise
6. Modify the Release Date to be after the old Start Date
7. Modify the Start Date to be after the old Start Date and before the new Release Date
8. Verify that the Quiz Exercise can be saved

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage



| Class/File | Branch | Line |
|------------|-------:|-----:|
| quiz-exercise-detail.component.ts | 79.71% | 95.08% |

### Screenshots


#### Before

https://user-images.githubusercontent.com/10885503/204447348-cba540f7-cefb-4364-9121-1e3a239a08fc.mov

#### After

https://user-images.githubusercontent.com/10885503/204447388-bbe0b4cf-9102-432c-8c5a-a62478cb2828.mov
